src/core/io.jl

```diff
@@ -175,6 +175,8 @@ function Base.getindex(c::Chain, v::Symbol)
   end
 end
 
+Base.getindex(c::Chain, expr::Expr) = c[Symbol(string(expr))]
```

Comment:
We need some sanity check here, e.g. verify ‘expr’ is an indexing expression in the form of ‘x[1]…’.